### ISSUE TITLE
[Accessibility]: Wildcard `<Icon />` accessibility

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#34582]
### Audit type

General/Other

### Problem description

Our `Icon` component is not currently accessible and will cause issues across our application.

Screen readers will report each icon as "unlabelled image" - which isn't helpful to anyone.

### Expected behavior

For decorative icons, we should set `aria-hidden="true"`. This will ensure screen readers completely ignore the icon, as it adds no value to the user journey.

For icons that add useful information to the page, we should set `role="img"` and `aria-label="Some descriptive text"`. This will ensure screen readers are able to properly understand to meaning of the icon.

### Additional details

Note: Related work here https://github.com/sourcegraph/sourcegraph/issues/32379.

Ideally we could use the built in `<title>` and `<description>` attributes within our SVG elements. Unfortunately that will require migrating to a different icon library which will require more work and a codemod to do so efficiently.

We're focusing on just the accessibility aspect right now, once we have that fixed we can look into migrating to a new library.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [ ] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by GitStart. I have assigned the label 'gitstart' and added the assignee 'gitstart-sourcegraph'.

**Comment by umpox** [comment]
@gitstart-sourcegraph Please note this issue.

I think we can aim to solve this in three parts:

1. Add `role="img"` to `<Icon />` and fix all of the integration test issues. See this PR: https://github.com/sourcegraph/sourcegraph/pull/34578/files. Feel free to use that branch or copy the code from it - it just needs all the issues reported by `accessibilityAudit` fixed.
2. Migrate the rest of the codebase (split PRs into different areas)
3. Enforce for future `<Icon />` usage by enforcing either `title` or `aria-hidden` by making the props required [here](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@tr/icon-accessibility/-/blob/client/wildcard/src/components/Icon/Icon.tsx?L21-25&utm_source=VSCode-2.2.1) 

See the "Expected behavior" section of the issue for how we should decide between `aria-hidden` and `title`. It essentially boils down to "Does this icon help the user in understanding what they're trying to do?". LMK if there's any questions here.

**Comment by gitstart-sourcegraph** [comment]
## Related Issues
https://github.com/sourcegraph/sourcegraph/issues/34889
https://github.com/sourcegraph/sourcegraph/issues/34961

**Comment by umpox** [comment]
Now complete!
